scss/components/button.scss

```diff
@@ -90,20 +88,20 @@ a.#{$block} {
       @include fd-hover() {
           --fd-button-background-color: var(--fd-color-action-hover);
           --fd-button-color: var(--fd-color-action-2);
-          @include fd-var-color("color", fd-color("action",2));
-          @include fd-var-color("background-color", fd-color-state("hover","action"));
+          @include fd-var-color("color", fd-color("action",2), --fd-button-color);
+          @include fd-var-color("background-color", fd-color-state("hover","action"), --fd-button-background-color);
```

Comment:
The majority of the changes in this file are adding the additional argument for the mixin so all CSS variable references also include a default value.